src/backend/distributed/master/master_repair_shards.c

```diff
@@ -488,3 +511,44 @@ WorkerApplyShardDDLCommandList(List *ddlCommandList, int64 shardId)
 
 	return applyDdlCommandList;
 }
+
+
+/*
+ * LockReferencedTablesIfExist takes exclusive lock on the reference tables
+ * which has a foreign key from the given relation. This function is used
+ * for shard_rebalancer and tenant isolation to avoid modifications on the
```

Comment:
`shard_rebalancer ` and `tenant isolation` is not defined in this repo. So, we should not explicitly mention those here. And, I think we should have these comments at the point where we call this function.

Also, the name of the function is too generic. We should probably call it something like : `LockReferencedShardDistributionMetadata `.

Finally, for the future, it'd probably be less painful to maintain this function if we pass the `LockMode` as a parameter, instead of hard-coding `ExclusiveLock `